src/builtins.ts

```diff
@@ -771,6 +802,51 @@ export function compileCall(
         }
       }
     }
+    case BuiltinSymbols.nameof: {
+      // Check to make sure a parameter or a type was passed to the builtin
+      let resultType = evaluateConstantType(compiler, typeArguments, operands, reportNode);
+      if (resultType === null) {
+        compiler.currentType = compiler.program.stringInstance.type;
+        compiler.error(
```

Comment:
`evaluateConstantType` already reports in the `null` case.

Reply:
Got it.